### ISSUE TITLE
How to invoke (ValueChanged) callback and EditContext.NotifyFieldChanged from JSInterop?

### ISSUE THREAD

**Issue by Liero** [dotnet/aspnetcore#34703]
_I'm trying to reopen the issue #34532, as it was closed prematurely. Based on the answers, the the problem was not understood correctly, so I will give it one more chance._

I'm trying to implement CustomInput control, that will integrate with EditContext the similar to TextInput, but it will use JS interop call from Javascript to .NET instance.

```csharp
[JSInvokable]
public void IncrementJsInterop()  {
    Value++;
    //see see https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Forms/InputBase.cs#L79
    _ = ValueChanged.InvokeAsync(Value); 
    EditContext.NotifyFieldChanged(FieldIdentifier.Create(ValueExpression));
}
```

**The problem  is rerendering of the parent component**.

In this case, the parent is rerendered immediately after `ValueChanged` is invoked and it does not give the parent any chance to reflect changes made by `EditContext.NotifyFieldChanged()`

Normally, the parent rerender would be triggered only after the eventhandler is completed, that means after `EditContext.NotifyFieldChanged()` is called. Is it possible to achieve this using JSInvokable?

**Comment by Liero** [comment]
Closing in favor #42686 42686
